flagsmith/flagsmith.py

```diff
@@ -124,20 +140,54 @@ def __init__(
                         "in the environment settings page."
                     )
 
-                self.environment_data_polling_manager_thread = (
-                    EnvironmentDataPollingManager(
+                if self.use_stream:
+                    self.update_environment()
```

Comment:
Do we want a blocking operation here? I do understand that getting rid of this will add an edge case.
@matthewelwell thoughts?

Reply:
I see what you mean. 
The main reason for `self.update_environment()` here is to get the `api_key` from `self._environment` for the stream url, which as far as I understand it is different from the `environment_key` passed in. If there is a different way of obtaining this that I've missed we could launch straight in.